service/gcsutils/gcstools/netnsConfig.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+// This utility moves a network interface into a network namespace and
+// configures it. The configuration is passed in as a JSON object
+// (marshalled prot.NetworkAdapter).  It is necessary to implement
+// this as a separate utility as in Go one does not have tight control
+// over which OS thread a given Go thread/routing executes but as can
+// only enter a namespace with a specific OS thread.
+//
+// Note, this logs to stdout so that the caller (gcs) can log the
+// output itself.
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/Microsoft/opengcs/service/gcs/prot"
+	log "github.com/Sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+)
+
+func netnsConfigMain() {
+	if err := netnsConfig(); err != nil {
+		log.Error("netnsConfig returned: ", err)
+	}
+	os.Exit(0)
+}
+
+func netnsConfig() error {
+	ifStr := flag.String("if", "", "Interface/Adapter to move/configure")
+	nspid := flag.Int("nspid", -1, "Process ID (to locate netns")
+	cfgStr := flag.String("cfg", "", "Adapter configuration (json)")
+
+	flag.Parse()
+	if *ifStr == "" || *nspid == -1 || *cfgStr == "" {
+		return fmt.Errorf("All three arguments must be specified")
+	}
+
+	var a prot.NetworkAdapter
+	if err := json.Unmarshal([]byte(*cfgStr), &a); err != nil {
+		return err
+	}
+
+	if a.NatEnabled {
+		log.Infof("Configure %s in %d with: %s/%d gw=%s", *ifStr, *nspid, a.AllocatedIPAddress, a.HostIPPrefixLength, a.HostIPAddress)
+	} else {
+		log.Infof("Configure %s in %s with DHCP", *ifStr, *nspid)
+	}
```

Comment:
I wanted this output at the start, because otherwise, when something errors inbetween this and the other log output I don't know what the config was.